meeting-notes/2019/Q1/2019-01-07--js-core-dev-team-weekly.md

```diff
@@ -0,0 +1,160 @@
+- **Notetaker:** @jacobheun
+- **Attendees:**
+  - @vasco-santos
+  - @zcstarr
+  - @litzenberger
+  - @vmx
+  
+- **Recording:** https://youtu.be/q8evwA86OnI
+
+## Agenda
+
+- Ask everyone to put their name into the list of attendees
+- Round of updates
+  - What have you accomplished since the last Weekly?
+  - Were there any blockers? If so, which ones? Is it still blocked? Why?
+  - What is the next important thing you should focus on?
+- Ask for general questions. Could be things like:
+  - I'm stuck with something, I don't know who to ask. Who knows who to ask?
+  - Who can help me with xyz?
+- Plan this week
+  - Select issues to work on
+- Review remaining issues if there is time left
+
+## Notes
+
+### Week Update
+
+@alanshaw (not attending)
+- Done:
+    - PR review for [DHT interop tests](https://github.com/ipfs/interop/pull/36)
+    - PR review for [IPNS pubsub interop tests](https://github.com/ipfs/interop/pull/39)
+    - PR review for [IPNS over DHT interop tests](https://github.com/ipfs/interop/pull/47)
+    - Looked into [slow `cat` issue](https://github.com/ipfs/js-ipfs/issues/1706)
+        - We need an streaming `ipld.getMany` - see [conversation here](https://github.com/ipfs/js-ipfs-unixfs-exporter/pull/13)
+        - [Opened PR to fix initial 10s delay of boostrap node discovery](https://github.com/libp2p/js-libp2p-bootstrap/pull/86)
+    - Spent time familiarising myself with `libp2p-switch`, `libp2p-mplex` and `multistream-select` - I've started putting together a proposal for `libp2p-mplex` with async iterators
+    - Reviewed perf PR for `datastore-fs` to replace `write-file-atomic` instead of `fast-write-atomic`
+        - Reviewed `fast-write-atomic` code, opened 2 issues [#3](https://github.com/mcollina/fast-write-atomic/issues/3) and [#4](https://github.com/mcollina/fast-write-atomic/issues/3) which were resolved quickly by @mcollina
+    - Helped with reviews and releases necessary for resolution of [stack overflow issue](https://github.com/libp2p/js-libp2p-switch/issues/287)
+    - Volunteered to [start a ProtoSchool chapter in London](https://github.com/ProtoSchool/organizing/issues/3)
+- Blocked:
+    - 0.34 release now blocked on an [issue handling control signals](https://ci.ipfs.team/blue/organizations/jenkins/IPFS%2Fjs-ipfs/detail/PR-1803/3/pipeline/257) (also in master 😱)
+        - I am on holiday this week - would appreciate someone looking into this!
+- Next:
+    - n/a - on holiday
+
+@vasco-santos
+- Done:
+  - IPNS spec ready, reviewed and merged [ipfs/specs#184](https://github.com/ipfs/specs/pull/184)
+  - IPNS / DHT interop reviews by Alan were resolved [ipfs/interop#36](https://github.com/ipfs/interop/pull/36), [ipfs/interop#39](https://github.com/ipfs/interop/pull/39) and [ipfs/interop#47](https://github.com/ipfs/interop/pull/47)
+  - JS libp2p-daemon-client with basic commands and dht almost ready [libp2p/js-libp2p-daemon-client#1](https://github.com/libp2p/js-libp2p-daemon-client/pull/1)
```

Comment:
Fantastic! One thing that was unclear to me was https://github.com/libp2p/js-libp2p-daemon-client/pull/1/files#r245892243

If it is a real client (accessing remotely), which RPC library are you using?

Reply:
This uses unix sockets for communication between processes executing on the same host (through protobuf)

More information in the [spec](https://github.com/libp2p/go-libp2p-daemon/blob/master/specs/CONTROL.md#technical-details) and [roadmap](https://github.com/libp2p/go-libp2p-daemon/blob/1d7d0a25b6152441e4e8a216d53ee2f7240a161e/ROADMAP.md#introduction)